server/service/src/sensor/berlinger.rs

```diff
@@ -143,87 +114,98 @@ fn sensor_add_breach_if_new(
     sensor_row: &SensorRow,
     temperature_breach: &temperature_sensor::TemperatureBreach,
     breach_config: &temperature_sensor::TemperatureBreachConfig,
-) -> Result<(), RepositoryError> {
+) -> Result<Option<TemperatureBreachRow>, RepositoryError> {
     let breach_row_type = get_breach_row_type(&temperature_breach.breach_type);
-    let result = get_matching_sensor_breach(
+    let temperature_breach_option = get_matching_sensor_breach(
         connection,
         &sensor_row.id,
         temperature_breach.start_timestamp,
-        temperature_breach.end_timestamp,
         &breach_row_type,
     )?;
 
-    if let Some(mut record) = result.clone().pop() {
```

Comment:
Some of the changes here are just stylistic prefs, but I think they are more consistent with existing code, but could be wrong and it's just my opinionated change, sorry